Jenkinsfile

```diff
@@ -0,0 +1,68 @@
+def nodes = ['linux']
+def release_modes = ['debug', 'optimized', 'release']
+def release_args = ['--debug', '--O3', '--release-with-symbols']
+
+def CFLAGS = "CFLAGS=\"-m64\""
+def LDFLAGS = "LDFLAGS=\"-m64\""
+def DEFAULT_FLAGS = "${CFLAGS} ${LDFLAGS}"
+def tasks = [:]
+
+for(int i = 0; i < nodes.size(); i++)
+{
+    def node_is = nodes[i]
+
+    for(int j = 0; j < release_modes.size(); j++)
+    {
+        def name = release_modes[j]
+        def option = release_args[j]
+
+        // By default, builds here only test for successful compilation.
+        def test_cmd = 'calacs.e --version'
+
+        // One of them should run "remote_data" tests when in release mode.
+        // "slow" tests should only run in nightly build and not addressed here.
+        if (name == 'release') {
+            test_cmd = 'pytest tests --basetemp=tests_output --junitxml results.xml --remote-data'
+        }
+
+        tasks["${node_is}/${name}"] = {
+            node {
+                stage("Checkout") {
+                    checkout scm
+                }
+
+                def prefix = pwd() + '/_install'
+                def runtime = ["PATH=${prefix}/bin:${env.PATH}"]
+
+                stage("Generate (${name})") {
+                    sh "yes '' | ./waf configure --prefix=${prefix} ${option} ${DEFAULT_FLAGS}"
+                    sh './waf build'
+                    sh './waf install'
+                }
+                try {
+                    stage("Test (${name})") {
+                        if (name == 'release') {
+                            sh 'conda install -q -y pytest requests astropy'
+                            sh 'pip install -q pytest-remotedata'
+                        }
+                        withEnv(runtime) {
+                            sh "${test_cmd}"
+                        }
+                    }
+                }
+                catch (e) {
```

Comment:
I know this was based on what was in my own Jenkinsfile, but I just wanted to make sure you know:
`catch (e) {` _only_ records test information in the event of a failure. If you replace with this with `finally {` the test results will be recorded no matter what happens.

Reply:
The report as-is is good enough for me. And in previous attempt with failure, I found that info reported as sufficient for tracking where the failure came from as well (at least for that case). Thank you!

```
+ pytest tests --basetemp=tests_output --junitxml results.xml --remote-data
============================= test session starts ==============================
platform linux -- Python 3.6.1, pytest-3.2.1, py-1.4.34, pluggy-0.4.0

rootdir: ..., inifile:
plugins: remotedata-0.2.0
collected 15 items

tests/acs/test_hrc.py .
tests/acs/test_sbc.py s.
tests/acs/test_wfc.py .
tests/stis/test_lev1_fuvspec.py x
tests/utils/test_helpers.py ........s
tests/wfc3/test_ir_01.py .

 generated xml file: .../results.xml 

============== 12 passed, 2 skipped, 1 xfailed in 239.93 seconds ===============
```

Reply:
Sounds good. To take full advantage of the analyzer's pass/fail/diff/newtest/oldtest graphing, drill downs, and "weather report" functionality you'd probably need to keep a steady stream of results. Otherwise you'd only be diffing failures over time.

Reply:
> diffing failures over time

Ah, that would be bad. I'll change it to `finally` as you suggested. Thank you!